Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -135,5 +140,81 @@ public void RespectAnnotationPrefixes()
                 "my.universal.namespace.BussyMcBusface", 
                 new IonSerializer().Serialize(new Bus()));
         }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToParentClassNameMatchingAnnotation()
+        {
+            string truckIonText = "Truck:: { }";
+
+            IIonReader reader = IonReaderBuilder.Build(truckIonText);
+
+            IonSerializer ionSerializer = new IonSerializer();
+            Vehicle truck = ionSerializer.Deserialize<Vehicle>(reader);
+
+            Assert.AreEqual(truck.ToString(), TestObjects.nativeTruck.ToString());
+        }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToClassNameMatchingAnnotation()
+        {
+            string truckIonText = "Truck:: { }";
+
+            IIonReader reader = IonReaderBuilder.Build(truckIonText);
+
+            IonSerializer ionSerializer = new IonSerializer();
+            Truck truck = ionSerializer.Deserialize<Truck>(reader);
+
+            Assert.AreEqual(truck.ToString(), TestObjects.nativeTruck.ToString());
+        }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToClassNameNotMatchingAnnotation()
+        {
+            string annotatedIonText = "NonMatching:: { }";
+
+            IIonReader reader = IonReaderBuilder.Build(annotatedIonText);
+
+            IonSerializer ionSerializer = new IonSerializer();
+            Truck truck = ionSerializer.Deserialize<Truck>(reader);
+
+            Assert.AreEqual(truck.ToString(), TestObjects.nativeTruck.ToString());
+        }
+
+        [TestMethod]
+        public void DeserializeAnnotatedIonToClassNameWithAnnotatedTypeAssemblies()
+        {
+            IonSerializationOptions options = new IonSerializationOptions
+            {
+                AnnotatedTypeAssemblies = new string[]
+                {
+                        typeof(Truck).Assembly.GetName().Name
```

Comment:
I think proof by counterexample would be more effective here since this will continue to work even if you remove this option.

Reply:
Logic has been changed to not search all loaded assemblies if a list of assemblies are given.